src/main/kotlin/lotto/domain/PrizePolicy.kt

```diff
@@ -0,0 +1,5 @@
+package lotto.domain
+
+interface PrizePolicy<LottoTicket, PrizeType> {
+    fun won(lottoTicket: LottoTicket): PrizeType?
+}
```

Comment:
LottoTicket 로 PrizeType 라는 상을 준다라는 개념을 표현하기 위해 제네릭을 사용하신것 같은데요!
음 저는 제네릭을 사용할 필요성이 있는지 잘 모르겠습니다.
```suggestion
interface PrizePolicy {
    fun won(lottoTicket: LottoTicket): Money?
}
```

지금은 위와 같이 되어도 충분할것 같고
정말로 제네릭을 사용하고 싶으시다면
```suggestion
interface PrizePolicy<PrizeType> {
    fun won(lottoTicket: LottoTicket): PrizeType?
}
```
와 같이 사용하는것이 될것 같습니다.

두가지 모두 제네릭을 사용하면 이 정책이 너무나도 추상적이게 되는것 같습니다.
저는 명확한 정책의 성격을 띄고 있어주는게 좋다고 생각합니다!

Reply:
원래 당첨 조건에 상품이 현금 만이 아닌 다른 상품도 넣을수 있게 설계 하기 위해 인터페이스를 설정 하였는데
과제 하면서 점점 코드가 어렵게 되는 것 같아서 제거 하였습니다 ~